pkg/operator/controller/controller_dns_daemonset_test.go

```diff
@@ -469,14 +481,23 @@ func TestDaemonsetConfigChanged(t *testing.T) {
 			expect: true,
 		},
 		{
-			description: "if the update strategy changes",
+			description: "if the update strategy's max unavailable parameter changes",
 			mutate: func(daemonset *appsv1.DaemonSet) {
-				daemonset.Spec.UpdateStrategy = appsv1.DaemonSetUpdateStrategy{
-					Type: appsv1.RollingUpdateDaemonSetStrategyType,
-					RollingUpdate: &appsv1.RollingUpdateDaemonSet{
-						MaxUnavailable: pointerTo(intstr.FromString("10%")),
-					},
-				}
+				daemonset.Spec.UpdateStrategy.RollingUpdate.MaxUnavailable = pointerTo(intstr.FromString("10%"))
+			},
+			expect: true,
+		},
+		{
+			description: "if the update strategy's max surge parameter changes",
+			mutate: func(daemonset *appsv1.DaemonSet) {
+				daemonset.Spec.UpdateStrategy.RollingUpdate.MaxSurge = pointerTo(intstr.FromString("10%"))
```

Comment:
This is not a change, is it?

Reply:
I guess it is. I expected that you would have used the manifest values as the base, but upon further checking, you didn't.